### PR TITLE
Added SequencedCollection interface, inherited List from it

### PR DESCRIPTION
**Pull request by Ihromant**
Added SequencedCollection interface, inherited List from it (initial part of JEP-431).

Can't properly cover it by tests (JDK-21 is not build JDK for teavm now), but it should work according to the common logic. Not reimplemented ListIterator and Iterator for reversed(), so they should not work properly for now (need tests to do it). Other reversed() overridden methods (and some not overridden) should work correctly because they have obvious implementation.

### CODE REVIEW

**Comment by konsoletyper** [comment]
> Can't properly cover it by tests (JDK-21 is not build JDK for teavm now)

But what's the problem to switch?

> Not reimplemented ListIterator and Iterator for reversed(), so they should not work properly for now (need tests to do it). 

Default implementations of `iterator`/`listIterator` in `AbstractList` utilize `get` and `size`, so whenever you implemented them properly, they also should work.

File: classlib/src/main/java/org/teavm/classlib/java/util/TList.java [hunk 1]
```diff
@@ -50,6 +50,94 @@ default void sort(TComparator<? super E> c) {
...
+    default E removeLast() {
+        if (isEmpty()) {
+            throw new TNoSuchElementException();
+        }
+        return remove(this.size() - 1);
+    }
+
+    @Override
+    default TList<E> reversed() {
+        return new TAbstractList<>() {
```
Comment (konsoletyper) [inline classlib/src/main/java/org/teavm/classlib/java/util/TList.java:97]: Other methods worth implementing here (only for performance reasons): addAll, clear, isEmpty, contains, addFirst, addLast, getFirst, getLast, removeFirst, removeLast, reversed, listIterator

**Comment by konsoletyper** [comment]
It's useful to implement these `*First`/`*Last` methods in `LinkedList`

**Comment by Ihromant** [comment]
@konsoletyper I suppose that implementation is finished, please review.

File: classlib/src/main/java/org/teavm/classlib/java/util/TList.java [hunk 2]
```diff
@@ -159,4 +208,78 @@ static <E> TList<E> of(E... elements) {
     static <E> TList<E> copyOf(TCollection<? extends E> collection) {
         return new TTemplateCollections.ImmutableArrayList<>(collection);
     }
+
+    class ReversedList<E> extends TAbstractList<E> {
```
Comment (konsoletyper) [inline classlib/src/main/java/org/teavm/classlib/java/util/TList.java:212]: I'd brought it to top level. And made it package-private

Comment (konsoletyper) [reply classlib/src/main/java/org/teavm/classlib/java/util/TList.java:212]: Implementing `iterator` would also be quite useful here

File: classlib/src/main/java/org/teavm/classlib/java/util/TReversedList.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.util;
+
+class TReversedList<E> extends TAbstractList<E> {
```
Comment (konsoletyper) [inline classlib/src/main/java/org/teavm/classlib/java/util/TReversedList.java:18]: getFirst/getLast too, please

File: classlib/src/main/java/org/teavm/classlib/java/util/TList.java (hunk 2 above)
Comment (Ihromant) [reply classlib/src/main/java/org/teavm/classlib/java/util/TList.java]: It uses implementation of iterator from AbstractList which is simple and performant.

Comment (konsoletyper) [reply classlib/src/main/java/org/teavm/classlib/java/util/TList.java]: It's not. Underlying list does not necessarily implement `get` in the most efficient ways. Sometimes it's much faster to iterate over list than getting elements with `get` method.

**Pull request merged**